### ISSUE TITLE
Rule to forbid fully qualified names

### ISSUE THREAD

**Issue by p-dmitriy** [detekt/detekt#8880]
## Expected Behavior of the rule

Rule that forbid to use full qualified names. 
Incorrect:
```kotlin
import io.internal.api.FileKind

fun createFile(
    name: kotlin.String,
    fileKind: io.internal.domain.FileKind,
): java.io.File {
    // ...
}
```
Correct:
```kotlin
import io.internal.api.FileKind
import io.internal.domain.FileKind as DomainFileKind

fun createFile(
    name: String,
    fileKind: DomainFileKind,
): File {
    // ...
}
```

## Context

Such code can be a result of refactoring with poor type inference from IDE or an inattention from developer when two classes with the same name are used. Alias import usage make this code more readable and more error proof to cases when incorrect class is used.

**Comment by forketyfork** [comment]
Looks the same as [#1501](https://github.com/detekt/detekt/issues/1501)

**Comment by cortinico** [comment]
> Looks the same as [#1501](https://github.com/detekt/detekt/issues/1501)

Indeed. Closing as duplicate of #1501 - let's keep the conversation in a single place
